pkg/cmd/flags/event.go

```diff
@@ -0,0 +1,119 @@
+package flags
+
+import (
+	"strings"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+)
+
+// PolicyEventMap maps policy id to its pre-parsed event flag fields
+type PolicyEventMap map[int]policyEvents
+
+// policyEvents holds pre-parsed event flag fields of one policy
+type policyEvents struct {
+	policyName string
+	eventFlags []*eventFlag
+}
+
+// eventFlag holds pre-parsed event flag fields
+type eventFlag struct {
+	full              string
+	eventFilter       string
+	eventName         string
+	eventOptionType   string
+	eventOptionName   string
+	operator          string
+	values            string
+	operatorAndValues string
+	filter            string
+}
+
+func PrepareEventMapFromFlags(eventsArr []string) (PolicyEventMap, error) {
+	// parse and store events flags
+	var evtFlags []*eventFlag
+	for _, evtFlag := range eventsArr {
+		parsed, err := parseEventFlag(evtFlag)
+		if err != nil {
+			return nil, err
+		}
+
+		evtFlags = append(evtFlags, parsed)
+	}
+
+	eventMap := make(PolicyEventMap)
+	eventMap[0] = policyEvents{eventFlags: evtFlags}
+
+	return eventMap, nil
+}
+
+// parseEventFlag parses an event flag and returns an eventFlag struct with
+// pre-parsed fields, or an error if the flag is invalid.
+func parseEventFlag(flag string) (*eventFlag, error) {
+	if flag == "" {
+		return nil, errfmt.WrapError(InvalidFlagEmpty())
+	}
+
+	operatorIdx := strings.IndexAny(flag, "=!<>")
+	if operatorIdx == -1 { // no operator
+		// "openat.context.container" edge case
+		if strings.Contains(flag, ".") {
+			evtParts, err := getEventFilterParts(flag, flag)
+			if err != nil {
+				return nil, errfmt.WrapError(err)
+			}
+
+			return &eventFlag{
+				full:            flag,
+				eventFilter:     flag,
+				eventName:       evtParts.name,
+				eventOptionType: evtParts.optType,
+				eventOptionName: evtParts.optName,
+			}, nil
+		}
+
+		if hasLeadingOrTrailingWhitespace(flag) {
+			return nil, errfmt.WrapError(InvalidFilterFlagFormat(flag))
+		}
+
+		return &eventFlag{
+			full:      flag,
+			eventName: flag,
+		}, nil
+	}
+
+	evtFilter := flag[:operatorIdx]
+	operatorAndValues := flag[operatorIdx:]
+
+	// validate event filter
+	evtParts, err := getEventFilterParts(evtFilter, flag)
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	filter := flag[len(evtParts.name)+1:]
+
+	// validate operator
+	operatorEndIdx := strings.LastIndexAny(operatorAndValues, "=!<>")
```

Comment:
Can it be that `operatorIdx` is different than `operatorEndIdx`? If so, can you give an example?

Reply:
If there's a operator greater than one char, so both operatorIdx and operatorEndIdx will be different. E.g.:

`a!=b`

`operatorIdx = 1`
`operatorEndIdx = 2`

Putting these comments.

```go
operatorIdx := strings.IndexAny(flag, "=!<>") // get first idx of any
operatorEndIdx := strings.LastIndexAny(operatorAndValues, "=!<>") // get last idx of any
```

Reply:
There's a bug which can wrongly consider a value char as the last index:

`something=some!how`

Fixing it.